homeassistant/components/vorwerk/vacuum.py

```diff
@@ -0,0 +1,348 @@
+        elif self._state["state"] == 4:
+            self._clean_state = STATE_ERROR
+            self._status_state = ERRORS.get(self._state["error"])
+
+        self._battery_level = self._state["details"]["charge"]
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def supported_features(self):
+        """Flag vacuum cleaner robot features that are supported."""
+        return SUPPORT_VORWERK
+
+    @property
+    def battery_level(self):
+        """Return the battery level of the vacuum cleaner."""
+        return self._battery_level
+
+    @property
+    def available(self):
+        """Return if the robot is available."""
+        return self._available
+
+    @property
+    def icon(self):
+        """Return specific icon."""
+        return "mdi:robot-vacuum-variant"
+
+    @property
+    def state(self):
+        """Return the status of the vacuum cleaner."""
+        return self._clean_state
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._robot_serial
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the vacuum cleaner."""
+        data = {}
+
+        if self._status_state is not None:
+            data[ATTR_STATUS] = self._status_state
+        if self._clean_time_start is not None:
+            data[ATTR_CLEAN_START] = self._clean_time_start
+        if self._clean_time_stop is not None:
+            data[ATTR_CLEAN_STOP] = self._clean_time_stop
+        if self._clean_area is not None:
+            data[ATTR_CLEAN_AREA] = self._clean_area
+        if self._clean_susp_charge_count is not None:
+            data[ATTR_CLEAN_SUSP_COUNT] = self._clean_susp_charge_count
+        if self._clean_susp_time is not None:
+            data[ATTR_CLEAN_SUSP_TIME] = self._clean_susp_time
+        if self._clean_pause_time is not None:
+            data[ATTR_CLEAN_PAUSE_TIME] = self._clean_pause_time
+        if self._clean_error_time is not None:
+            data[ATTR_CLEAN_ERROR_TIME] = self._clean_error_time
+        if self._clean_battery_start is not None:
+            data[ATTR_CLEAN_BATTERY_START] = self._clean_battery_start
+        if self._clean_battery_end is not None:
+            data[ATTR_CLEAN_BATTERY_END] = self._clean_battery_end
+        if self._launched_from is not None:
+            data[ATTR_LAUNCHED_FROM] = self._launched_from
+
+        return data
+
+    @property
+    def device_info(self):
+        """Device info for robot."""
+        info = {
+            "identifiers": {(VORWERK_DOMAIN, self._robot_serial)},
+            "name": self._name,
+        }
+        if self._robot_stats:
+            info["manufacturer"] = self._robot_stats["battery"]["vendor"]
+            info["model"] = self._robot_stats["model"]
+            info["sw_version"] = self._robot_stats["firmware"]
+        return info
+
+    def start(self):
```

Comment:
Can't this be async?

Reply:
I don't know. How can I check this? Any problem to keep it as is?

Reply:
I don't know. How can I check this? Any problem to keep it as is?

Reply:
It all depends on the underlying package, if that support async then it would make sense to change this to `async def async_start(self):` and then have await before the call into the package, this looks like it will block while it is waiting for a response!

Reply:
I will check it.

Reply:
Yes it is waiting for a response, but the underlying package doesn't support async. So for now I will leave it as it.